### PR TITLE
Feature/favourite coredata

### PR DESCRIPTION
**Pull request by AShozi**
core data functionality in accounts screen

### CODE REVIEW

File: phantomgames/Accounts/AccountsViewController.swift [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    
+    private func setupTableView() {
+        
+        favouriteTableView.register(FavoriteCustomTableViewCell.favoritetableViewNib(),
+                                    forCellReuseIdentifier: Constants.TableViewIdentifiers.customFavoriteCellIdentifier)
+        favouriteTableView.delegate = self
+        favouriteTableView.dataSource = self
+        
+    }
+
```
Comment (AShozi) [inline phantomgames/Accounts/AccountsViewController.swift]: fix spacing

File: phantomgames/Accounts/AccountsViewController.swift [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        favouriteTableView.reloadData()
+    }
+    
+    private func setupTableView() {
+        
+        favouriteTableView.register(FavoriteCustomTableViewCell.favoritetableViewNib(),
+                                    forCellReuseIdentifier: Constants.TableViewIdentifiers.customFavoriteCellIdentifier)
+        favouriteTableView.delegate = self
+        favouriteTableView.dataSource = self
+
```
Comment (AShozi) [inline phantomgames/Accounts/AccountsViewController.swift:43]: fix spacing

File: phantomgames/Accounts/CustomCells/FavoriteCustomTableViewCell.swift [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        
+    }
+    
+    // MARK: IBAction
+    
+    @IBAction private func deleteButtonTapped(_ sender: UIButton) {
+        deleteHandler?()
+    }
+    
+    @IBAction func gameButtonTapped(_ sender: UIButton) {
```
Comment (AShozi) [inline phantomgames/Accounts/CustomCells/FavoriteCustomTableViewCell.swift]: make private

File: phantomgames/CoreData/FavouriteCoreDataModel.swift [hunk 4]
```diff
@@ -0,0 +1,82 @@
+//
+//  CoreDataModel.swift
+//  phantomgames
+//
+//  Created by Aphiwe Shozi on 2024/05/29.
+//
+import UIKit
+import CoreData
+
+// MARK: - Enum
```
Comment (AShozi) [inline phantomgames/CoreData/FavouriteCoreDataModel.swift]: space after marks

File: phantomgames/CoreData/FavouriteCoreDataModel.swift [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+                newFavorite.icon = gameDetail.thumbnail
+                newFavorite.genre = gameDetail.genre
+                newFavorite.gameDescription = gameDetail.description
+                newFavorite.releaseDate = gameDetail.releaseDate
+                newFavorite.platform = gameDetail.platform
+                newFavorite.gameURL = URL(string: gameDetail.gameURL)
+                
+                try context.save()
+            } else {
+                print("Game already exists in favorites")
```
Comment (AShozi) [inline phantomgames/CoreData/FavouriteCoreDataModel.swift:57]: remove print statements

**Review by kieran-woodrow — APPROVED** [review]

**Review by simekanidev — APPROVED** [review]

**Review by rvanderwesthuizen — APPROVED** [review]

**Pull request merged**